### PR TITLE
Delete unused modals

### PR DESCRIPTION
**Pull request by juliatan**
# Description

Deleted the MentalHealthModal and DietStudyModal, along with corresponding links, since they are no longer in use.
Ticket: https://www.notion.so/joinzoe/Replacing-Modal-Components-9386b6fc74644ff2932b317a88286c1a

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

N/A

## Checklist (for submission)

- ~[ ] Analytics/tracking events have been added (if needed)~

## Checklist (for reviewers)

- [ ] Checked against Figma designs (if relevant)
- [ ] Checked that data has been successfully saved to the backend (if relevant)

## Out of scope and potential follow-up

None

### CODE REVIEW

**Comment by selvach** [comment]
Could you also delete any unused i18n translation strings, please!

**Comment by juliatan** [comment]
> Could you also delete any unused i18n translation strings, please!

Ah yes, definitely. Good catch, thank you.

**Comment by bionara** [comment]
> > Could you also delete any unused i18n translation strings, please!
> 
> Ah yes, definitely. Good catch, thank you.

they always catch me too! or should i say, i never catch them?!

**Review by bionara — DISMISSED** [review]
Happy to merge when locales removed :) nice one!

File: assets/lang/en.json [hunk 1]
```diff
@@ -1630,7 +1623,7 @@
       "card": "How helpful were these insights to you?",
       "feedback": "Care to share some more feedback?",
       "not-helpful": "Not helpful",
-      "very-helpful": "Very helpful" 
+      "very-helpful": "Very helpful"
```
Comment (aikewoody) [inline assets/lang/en.json:1626]: Nice spot, we should really sanitise these language files :)

**Review by aikewoody — APPROVED** [review]

**Pull request merged**